sdk/identity/identity/src/credentials/azurePowerShellCredential.ts

```diff
@@ -35,12 +35,15 @@ export function formatCommand(commandName: string): string {
  * If anything fails, an error is thrown.
  * @internal
  */
-async function runCommands(commands: string[][]): Promise<string[]> {
+async function runCommands(commands: string[][], timeout?: number): Promise<string[]> {
   const results: string[] = [];
 
   for (const command of commands) {
     const [file, ...parameters] = command;
-    const result = (await processUtils.execFile(file, parameters, { encoding: "utf8" })) as string;
+    const result = (await processUtils.execFile(file, parameters, {
+      encoding: "utf8",
+      timeout: timeout,
```

Comment:
Nit could also use `timeout` without the `timeout: timeout`

Reply:
```suggestion
      timeout,
```